network/src/peer_manager/mod.rs

```diff
@@ -427,27 +425,12 @@ where
             PeerManagerRequest::DisconnectPeer(peer_id, resp_tx) => {
                 // Send a CloseConnection request to NetworkProvider and drop the send end of the
                 // NetworkRequest channel.
-                if let Some((_, _, mut sender)) = self.active_peers.remove(&peer_id) {
-                    if let Err(close_err) = sender.push(
-                        ProtocolId::from_static(b"DisconnectPeer"),
-                        NetworkRequest::CloseConnection,
-                    ) {
-                        info!(
-                            "Failed to initiate connection close. CloseConnection request could not be delivered. Error: {:?}",
-                            close_err
-                        );
-                        if let Err(send_err) = resp_tx.send(Err(PeerManagerError::from(close_err)))
-                        {
-                            info!(
-                                "Failed to send connection close error. Error: {:?}",
-                                send_err
-                            );
-                        }
-                    } else {
-                        // Add to outstanding disconnect requests.
-                        self.outstanding_disconnect_requests
-                            .insert(peer_id, resp_tx);
-                    }
+                if let Some((_, _, sender)) = self.active_peers.remove(&peer_id) {
+                    // This should trigger a disconnect.
+                    drop(sender);
```

Comment:
wouldn't this implicitly happen at line 434?

Reply:
yeah, just wanted to be more explicit about where we drop connections.